src/System.IO.FileSystem/tests/FileStream/ReadWriteSpan.netcoreapp.cs

```diff
@@ -0,0 +1,192 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public abstract class FileStream_ReadWrite_Span : FileSystemTest
+    {
+        protected abstract FileStream CreateFileStream(string path, FileMode mode, FileAccess access = FileAccess.ReadWrite);
+
+        [Fact]
+        public void DisposedStream_ReadWrite_Throws()
+        {
+            var fs = CreateFileStream(GetTestFilePath(), FileMode.Create);
+            fs.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => fs.Read(new Span<byte>(new byte[1])));
+            Assert.Throws<ObjectDisposedException>(() => fs.Write(new Span<byte>(new byte[1])));
+        }
+
+        [Fact]
+        public void EmptyFile_Read_Succeeds()
```

Comment:
well strictly they're not all reads of an empty file, they're reads of zero bytes.